tests/test_requests.py

```diff
@@ -2540,6 +2540,28 @@ def test_json_encodes_as_bytes():
     assert isinstance(p.body, bytes)
 
 
+def test_data_ecodes_noniterables():
+    class Class:
+        pass
+
+    class ClassStringable:
+        def __str__(self):
+            return "🔥arbClassStringable🔥"
+
+    body = {
+        "string": "string",
+        "float": 0.01,
+        "int": 100,
+        "bool": True,
+        "None": None,
+        "plain_class": Class(),
+        "stringable_class": ClassStringable(),
+    }
+    p = PreparedRequest()
+    p.prepare(method="post", url="https://www.example.com/", data=body)
+    assert isinstance(p.body, str)
```

Comment:
you should also check that the data (e.g. keys) must be present or not.
Add a test showing that "None" key is indeed absent.

Reply:
```python

import requests

if __name__ == "__main__":
    url = "https://httpbin.org/post"
    data = {'test_none': None, 'xx': 'hello'}

    # requests completes
    resp = requests.post(url, data=data)
    print(resp.json())

```

```
{'args': {}, 'data': '', 'files': {}, 'form': {'xx': 'hello'}, 'headers': {'Accept': '*/*', 'Accept-Encoding': 'gzip, deflate, br', 'Content-Length': '8', 'Content-Type': 'application/x-www-form-urlencoded', 'Host': 'httpbin.org', 'User-Agent': 'python-requests/2.31.0', 'X-Amzn-Trace-Id': 'Root=1-6635da3e-494f137927479f5227ac558c'}, 'json': None, 'origin': '82.120.131.88', 'url': 'https://httpbin.org/post'}
```

`test_none` does not show up in result.